### ISSUE TITLE
Diff-util

### ISSUE THREAD

**Issue by artem-zinnatullin** [pushtorefresh/redom#17]

**Comment by artem-zinnatullin** [comment]
- [] Abstraction from RecyclerView Adapter, something like `fun setupRenderer(rv: RecyclerView): Renderer`, `interface Renderer { fun render(components: List<Component>): Disposable }`

**Comment by artem-zinnatullin** [comment]
We've implemented basic DiffUtil integration only to realize that no matter what we do it'll have major drawback — we can't properly compare user lambdas like `onClick = { … }`. For this reason we've thought of splitting onBind into two stages: `onBind(state)` and `onBind(callbacks)` so that DiffUtil would allow us to compare `state` and then we would always apply callbacks no matter what. Then we'll be able to use `getChangePayload` to send precise updates of the `state` to `onBind(state)`.

However, then we realized that our current `onBind` already does everything DiffUtil would do: compares individual state changes (both `areContentsTheSame` and `getChangePayload` features) and binds user lambdas while `viewTypes` allow RecyclerView to distinguish and reuse views & component pairs of proper types.
